packages/plugins/rum/src/index.ts

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019-Present Datadog, Inc.
+
+import type { PluginOptions, GetPlugins, GlobalContext, Logger } from '@dd/core/types';
+import { InjectPosition } from '@dd/core/types';
+import path from 'path';
+
+import { CONFIG_KEY, PLUGIN_NAME } from './constants';
+import { getInjectionValue } from './sdk';
+import type { OptionsWithRum, RumOptions, RumOptionsWithSdk } from './types';
+import { validateOptions } from './validate';
+
+export { CONFIG_KEY, PLUGIN_NAME };
+
+export const helpers = {
+    // Add the helpers you'd like to expose here.
+};
+
+export type types = {
+    // Add the types you'd like to expose here.
+    RumOptions: RumOptions;
+    OptionsWithRum: OptionsWithRum;
+};
+
+export const getPlugins: GetPlugins<OptionsWithRum> = (
+    opts: OptionsWithRum,
+    context: GlobalContext,
+    log: Logger,
+) => {
+    const plugins: PluginOptions[] = [];
+    // Verify configuration.
+    const options = validateOptions(opts, log);
+
+    // NOTE: These files are built from "@dd/tools/rollupConfig.mjs" and available in the distributed package.
+    if (options.sdk) {
+        // Inject the SDK from the CDN.
+        context.inject({
+            type: 'file',
+            // Using MIDDLE otherwise it's not executed in context.
+            position: InjectPosition.MIDDLE,
```

Comment:
What does `MIDDLE` mean exactly?

To my knowledge, RUM needs to be the 1st script executed to be able to track all resources properly

Reply:
From the [README](https://github.com/DataDog/build-plugins/tree/yoann/v2.5.0/packages/plugins/injection#injection-plugin-):

> `InjectPosition.MIDDLE`: Added at the beginning of the entry file, within the context of the bundle.

So it's the first piece of code in the context of the bundle.
It would be similar to a NPM installation and usage I believe.

Reply:
To note the diff with `START`:

> `InjectPosition.START`: Added at the very beginning of the bundle, outside any closure.

Meaning it won't bundle anything with it (dependencies for instance). Basically `START` acts like the Banner concept in the bundling world.